### PR TITLE
Lrn it2 game stats

### PR DESCRIPTION
**Pull request by LancePants97**
Wrote methods for highest_total_score, lowest_total_score, percentage_home_wins, percentage_visitor_wins, and percentage_ties. I also adjusted the corresponding tests for these methods as well as adding one more line to the games csv fixture file. Still need to add tests for the helper methods I created.

### CODE REVIEW

File: lib/game_stats.rb [hunk 1]
```diff
@@ -1,4 +1,94 @@
 
 
+
+
 module GameStats
+
+	def highest_total_score(season) # integer
```
Comment (cheeleertr) [inline lib/game_stats.rb:7]: Nice and clean. Good use of helper methods.

File: lib/game_stats.rb [hunk 2]
```diff
@@ -1,4 +1,94 @@
...
+		home_wins_percentage.round(2) 
+	end
+
+	def percentage_visitor_wins(season) # float
+		away_winners(season)
+		games_float = @games.count.to_f
+		wins_float = @away_wins.count.to_f
+		home_wins_percentage = wins_float / games_float
+		home_wins_percentage.round(2) 
+	end
```
Comment (cheeleertr) [inline lib/game_stats.rb:61]: Interesting way of using home_winners and away_winners. I was wondering where @away_wins and @home_wins came from but they were defined when home_winners and away_winners methods were called.

File: lib/game_stats.rb [hunk 3]
```diff
@@ -1,4 +1,94 @@
...
+
+	def lowest_total_score(season)
+		if away_winners(season) < home_winners(season)
+			away_winners(season)
+		elsif away_winners(season) > home_winners(season)
+			home_winners(season)
+		end
+	end
+
+	def get_games(season) # helper
```
Comment (cheeleertr) [inline lib/game_stats.rb:23]: I might have missed something but I don't think #get_games was used anywhere.

File: lib/game_stats.rb [hunk 4]
```diff
@@ -1,4 +1,94 @@
...
+		@all_games = @games
+	end
+
+	def home_winners(season) # helper
+		@home_wins = []	
+			@games.each do |game|
```
Comment (jdmchugh111) [inline lib/game_stats.rb:29]: definitely a nit picky thing, but I don't think you need to tab in here - only when you're starting a new code block like under a def, it-do block, conditional or enumerator

File: lib/game_stats.rb (hunk 1 above)
Comment (jdmchugh111) [reply lib/game_stats.rb:7]: I'm seeing that you're passing in seasons as an argument for your methods. I'm not sure if you maybe looked at my code as a model, but I believe only the Season Stats uses season as a parameter. In this Game Stats section I think we should be creating a method that looks at the entire set of data, not just a specific season

File: lib/game_stats.rb [hunk 5]
```diff
@@ -1,4 +1,94 @@
...
+	def highest_total_score(season) # integer
+		if away_winners(season) > home_winners(season)
+			away_winners(season)
+		elsif away_winners(season) < home_winners(season)
+			home_winners(season)
+		end
+	end
+
+	def lowest_total_score(season)
+		if away_winners(season) < home_winners(season)
```
Comment (jdmchugh111) [inline lib/game_stats.rb:16]: Ok, so unfortunately I think you may have coded for the wrong stat on highest_total_score and lowest_total_score. Not sure if you saw Austin K's question to Kat in code-help a few days ago. But these methods should be looking at every game and determining the sums of home goals and away goals (if the score was 3-1, the "total score" would be 4). We can assume lowest_total_score would probably return 0 (from a 0-0 tie), but highest total score would probably be an integer around 10 or so

File: lib/game_stats.rb [hunk 6]
```diff
@@ -1,4 +1,94 @@
...
+			@games.each do |game|
+				if game.away_goals.to_i > game.home_goals.to_i	
+					@away_wins << game.away_goals.to_i
+				end
+			end
+		@away_wins.sum
+	end	
+
+	def percentage_home_wins(season) # float
+		home_winners(season)
```
Comment (jdmchugh111) [inline lib/game_stats.rb:48]: same thing here as above - I don't think we want season as a parameter. We should be looking at the entire data set and determining what percentage of all games was home wins, what percentage was away wins, and what percentage was ties

File: lib/game_stats.rb (hunk 2 above)
Comment (JCL461437) [reply lib/game_stats.rb:61]: Like Chee said, interesting application of the away wins and home wins data sets.

**Review by JCL461437 — APPROVED** [review]
Great work Lance! It looks like we will have a merge conflict. Negotiating that will be valuable. Excited to see what other additions you'll make in the future!

**Review by jdmchugh111 — COMMENTED** [review]
great work overall Lance! sorry if it seemed like I was ripping on your code, that was not my intention. Just a few of those things I made comments on jumped out to me when I was looking through, but definitely will be a pretty quick fix on the percentages, and with the highest and lowest scores I feel like that's more of an issue with the vague wording of the spec and not with your coding (all of the logic looks great) - whoever wrote the spec of it2 wasn't clear enough on defining what those methods should be analyzing

File: lib/game_stats.rb (hunk 3 above)
Comment (LancePants97) [reply lib/game_stats.rb:23]: You're right, this helper method was made at the beginning but I ended up not using it so I will get rid of it once I'm sure all the other methods are working the way theyre intended to.

File: lib/game_stats.rb (hunk 4 above)
Comment (LancePants97) [reply lib/game_stats.rb:29]: Good call, that was an accidental extra tab. Will fix this when I go back to fix stuff today!

File: lib/game_stats.rb (hunk 5 above)
Comment (LancePants97) [reply lib/game_stats.rb:16]: Good call, also thanks for explaining this to me over zoom. I totally misinterpreted those methods when reading the directions but will rework them today to get them working the way they are supposed to.

**Pull request closed without merging**